community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/slurmsync.py

```diff
@@ -522,6 +529,11 @@ def main():
     except Exception:
         log.exception("failed to sync custom scripts")
 
+    try:
+        update_topology(lkp)
```

Comment:
We are now generating topology on every sync. Description says "no drawbacks since no changes expected".

1. What is cost of generating topology on large cluster, say 5000 nodes. Even though we may not call reconfigure on every sync could this impact sync run time? Should we guard this behind a flag? For example when topology/none is selected then this does not run?
2. Why do we expect it not to change? Probably just a lack of understanding of our current state but seems like whole reason for this feature is we expect it to change.

Reply:
> What is cost of generating topology on large cluster, say 5000 nodes.

It's super (cheap < `0.01 sec`) currently, without using `physicalHost`

> Should we guard this behind a flag? For example when topology/none is selected then this does not run?

Good call, added condition to do it IFF `topology/tree` is used.

> Why do we expect it not to change?

Because only possible change (until we use `physicalHost`) is addition/removal of nodes, that in order should be cased by changes in configuration -> full blown reconfigure will happen before `update_topology`